### PR TITLE
Add JuliaFormatter and format

### PR DESCRIPTION
**Pull request by Affie**
This is only changes from automatically formatting with `format("./src/"; verbose=true)` and a format config file.

I was hoping this wouldn't break anything. It's too much changes to check now.

### CODE REVIEW

**Review by dehann — APPROVED** [review]

**Comment by dehann** [comment]
oops, only noticed the do not merge label after merging, sorry! I did review every line and looked good.

**Comment by Affie** [comment]
Ok, If you reviewed every line that's ok then. I had some trouble with the formatter breaking code and I tuned the settings somewhat. And didn't have time to go through all the lines. 
We will need to do some of the guidelines manually.

**Comment by dehann** [comment]
yeah,sorry about that.  Local tests on new master still pass (similar to CI):
```julia
[ Info: CSM-5 Clique 5 finished, solveKey=default
[ Info: Finished tree based init-ference
Test Summary:                 | [ Info: monitorCSMs: all tasks done
Pass  Total   Time
testing dead reckoning tether |    9      9  15.7s
     Testing IncrementalInference tests passed 

(@v1.8) pkg> 

```

And would be good to tweak one or two of the Formatter rules, but this is a huge improvement already thank you!

**Pull request merged**
